RIOT-OS/RIOT
# sys/auto_init: custom auto-initialization sequence

### Contribution description



This PR proposes a way to

1. reorder the RIOT auto-init module sequence
2. insert custom auto-init modules

using `Kconfig` and `XFA`.

This could also be useful for some custom boards which must setup some stuff that cannot be done in `board_init()`.
  
### Testing procedure



`tests/auto_init_custom`:
inserts two custom modules A and B that are initialized after `ztimer`.

See also `tests/auto_init_custom/bin/<BOARD>/tests_auto_init_custom.map`
and search for "auto_init_xfa" to look at the auto-init cross file array.

Add `#define ENABLE_DEBUG 1` in `auto_init.c` to print the auto-init sequence of modules.

### Issues/PRs references



See also #16529.
